OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -209,6 +209,9 @@ public bool HasAnyValidWeapons(Target t, bool checkForCenterTargetingWeapons = f
 			// PERF: Avoid LINQ.
 			foreach (var armament in Armaments)
 			{
+				if (armament.IsTraitDisabled)
+					continue;
```

Comment:
IMO @MustaphaTR's version is more readable, since it makes it clear that this part of the same validity check, and not something distinct.  Would you mind adjusting this?